packages/grpc-native-core/src/client.js

```diff
@@ -380,19 +380,18 @@ function Client(address, credentials, options) {
   options['grpc.primary_user_agent'] += 'grpc-node/' + version;
 
   // Resolve interceptor options and assign interceptors to each method
-  var interceptor_providers = options.interceptor_providers || [];
-  var interceptors = options.interceptors || [];
-  if (interceptor_providers.length && interceptors.length) {
+  if (_.isArray(options.interceptor_providers) && _.isArray(options.interceptors)) {
```

Comment:
Something tells me that the original intend of the code was to basically be `_.isArrayLike`, by checking if it had the `length` property.

Reply:
Also, unrelated: shouldn't we be doing something like:

```
if (!_.isArrayLike(options.interceptor_providers) && _.isSet(options.interceptor_providers)) {
  throw new client_interceptors.InterceptorConfigurationError('interceptor_providers needs to behave like an array')
}
```